vita3k/config/src/config.cpp

```diff
@@ -118,12 +118,12 @@ ExitCode serialize_config(Config &cfg, const fs::path &output_path) {
 ExitCode init_config(Config &cfg, int argc, char **argv, const Root &root_paths) {
     // Always generate the default configuration file
     Config command_line{};
-    serialize_config(command_line, root_paths.get_base_path() / "data/config/default.yml");
-    // Load base path configuration by default; otherwise, move the default to the base path
-    if (fs::exists(check_path(root_paths.get_base_path())))
-        parse(cfg, root_paths.get_base_path(), root_paths.get_base_path());
+    serialize_config(command_line, root_paths.get_pref_path() / "data/config/default.yml");
+    // Load config path configuration by default; otherwise, move the default to the pref path
+    if (fs::exists(check_path(root_paths.get_config_path())))
+        parse(cfg, root_paths.get_config_path(), root_paths.get_pref_path());
```

Comment:
I think the serialize_config call to pref_path is a mistake. base_path/data/config/default.yml is a read-only file containing the default config.

About using config_path instead of base_path for config, indeed that needs to be done for linux as in some cases, the base path is a read-only folder. On Windows, we would just have config_path = base_path.

Reply:
Ok, nevermind, serialize_config actually writes to the file. That doesn't make a lot of sense to be honest.

Reply:
i good see code, and see said
"serialize_config(command_line, root_paths.get_pref_path() / "data/config/default.yml");"
pref path no have this file, is inside base_path

Reply:
You can see in the github repository that there is no file config/default.yml in the data folder. This file is created by the app itself (and shouldn't need to be created honestly). Because base_path can be readonly it can't be in the base_path

Reply:
hmm if is read only using other can read 🤔 
can using exemple config folder 🤔 
but no pref path, is not for this

Reply:
I agree that using your definition that pref path is only vita files, it shoudn't indeed be in pref path.

Reply:
line here is keep wrong
>parse(cfg, root_paths.get_config_path(), root_paths.get_pref_path());

no have pref_path writed here